scpproxy/src/main.rs

```diff
@@ -0,0 +1,223 @@
+extern crate clap;
+use clap::{App, Arg};
+use dicom_ul::pdu::reader::{read_pdu, DEFAULT_MAX_PDU};
+use dicom_ul::pdu::writer::write_pdu;
+use dicom_ul::pdu::PDU;
+use quick_error::quick_error;
+use std::net::{Shutdown, TcpListener, TcpStream};
+use std::sync::mpsc;
+use std::sync::mpsc::{Receiver, Sender};
+use std::thread;
+use std::thread::JoinHandle;
+
+type Result<T> = std::result::Result<T, Error>;
+
+quick_error! {
+    #[derive(Debug)]
+    pub enum Error {
+        Io(err: std::io::Error) {
+            from()
+        }
+        DUL(err: dicom_ul::error::Error) {
+            from()
+        }
+        ThreadPanicked {
+            from()
+        }
+        RecvError(err: std::sync::mpsc::RecvError) {
+            from()
+        }
+        SendError(err: std::sync::mpsc::SendError<ThreadMessage>) {
+            from()
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum ProviderType {
+    SCP,
+    SCU,
+}
+
+#[derive(Debug)]
+pub enum ThreadMessage {
+    SendPDU {
```

Comment:
Although I'm willing to make exceptions to the [usual naming rules](https://rust-lang-nursery.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case) in some places, this is one place where I would say it's fine to stick to the convention: enum variants in UpperCamelCase, even for acronyms.

```suggestion
    SendPdu {
```